servers/rendering/shader_converter.cpp

```diff
@@ -0,0 +1,2591 @@
+	// Peek back until we hit a newline or the start of the file (EOF).
+	TokenE *start_pos = p_pos;
+	if (!start_pos) {
+		return false;
+	}
+	if (!_add_to_report(start_pos->get().line, p_comment, warning ? 1 : 0)) {
+		// Already added.
+		return true;
+	}
+	while (start_pos->prev() && start_pos->get().type != TT::TK_NEWLINE && start_pos->get().type != TT::TK_EOF) {
+		start_pos = start_pos->prev();
+	}
+	String block_comment = vformat("/* !convert%s */\n", (warning ? " WARNING: " : ": ") + p_comment);
+	// In case this has been run through the converter before, check if the token before this is a block comment and has the same comment.
+	TokenE *prev = start_pos->prev();
+	while (prev && prev->get().type == TT::TK_BLOCK_COMMENT) {
+		if (get_token_literal_text(start_pos->next()->get()).strip_edges() == block_comment.strip_edges()) {
+			return true;
+		}
+		prev = prev->prev();
+	}
+	return insert_after(mkTok(TT::TK_BLOCK_COMMENT, block_comment), start_pos);
+}
+
+bool ShaderDeprecatedConverter::_add_comment_at_eol(const String &p_comment, List<Token>::Element *p_pos) {
+	// Peek forward until we hit a newline or the end of the file (EOF).
+	TokenE *start_pos = p_pos ? p_pos : get_pos();
+	if (!start_pos) {
+		return false;
+	}
+	while (start_pos->get().type != TT::TK_NEWLINE && start_pos->get().type != TT::TK_EOF) {
+		start_pos = start_pos->next();
+	}
+	String comment = "/* !convert: " + p_comment + " */";
+	if (start_pos->prev() && start_pos->prev()->get().type == TT::TK_BLOCK_COMMENT && get_token_literal_text(start_pos->prev()->get()) == comment) {
+		return true;
+	}
+	return insert_before(mkTok(TT::TK_BLOCK_COMMENT, comment), start_pos);
+}
+
+void ShaderDeprecatedConverter::reset() {
+	ShaderLanguage sl;
+	code_tokens.clear();
+	sl.token_debug_stream(old_code, code_tokens, true);
+	code_tokens.push_back(eof_token);
+	code_tokens.push_front(eof_token);
+	uniform_decls.clear();
+	var_decls.clear();
+	struct_decls.clear();
+	function_decls.clear();
+	scope_declarations.clear();
+	after_shader_decl = code_tokens.front();
+	curr_ptr = code_tokens.front();
+	report.clear();
+}
+
+#define COND_MSG_FAIL(m_cond, m_msg) \
+	if (unlikely(m_cond)) {          \
+		err_str = m_msg;             \
+		return false;                \
+	}
+#define COND_LINE_MSG_FAIL(m_cond, m_line, m_msg) \
+	if (unlikely(m_cond)) {                       \
+		err_line = m_line + 1;                    \
+		err_str = m_msg;                          \
+		_add_to_report(err_line, err_str, 2);     \
+		return false;                             \
+	}
+#define LINE_MSG_FAIL(m_line, m_msg) \
+	err_line = m_line + 1;           \
+	err_str = m_msg;                 \
+	return false;
+#define MSG_FAIL(m_msg) \
+	err_str = m_msg;    \
+	return false;
+
+#define EOF_FAIL(m_tok_E)                                             \
+	COND_MSG_FAIL(m_tok_E == nullptr, RTR("Unexpected end of file")); \
```

Comment:
```suggestion
	COND_MSG_FAIL(m_tok_E == nullptr, RTR("Unexpected end of file.")); \
```
For all the custom errors, errors should be complete sentences and end with a period